paimon-codegen/src/main/java/org/apache/paimon/codegen/CodeGeneratorImpl.java

```diff
@@ -68,4 +148,40 @@ private SortSpec getAscendingSortSpec(int[] sortFields) {
         }
         return builder.build();
     }
+
+    /** Class to use as key for the {@link #GENERATED_CLASS_CACHE}. */
+    static class ClassKey {
+        private final String name;
```

Comment:
Should contains interface class, for example, `RecordEqualiser.class`.

Please add test for this.